service/src/main/java/com/derpgroup/derpwizard/manager/AbstractManager.java

```diff
@@ -64,6 +82,31 @@ public void handleRequest(@NonNull VoiceInput voiceInput, @NonNull SsmlDocumentB
     }
   }
 
+  private void appendToConversationHistory(@NonNull VoiceInput voiceInput) {
+    CommonMetadata metadata = voiceInput.getMetadata();
+    List<ConversationHistoryEntry> conversationHistory = metadata.getConversationHistory();
+    if(conversationHistory == null){
+      conversationHistory = new ArrayList<ConversationHistoryEntry>();
+    }
+    CommonMetadata metadataClone = null;
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      mapper.registerModule(mapperModule);
+      String metadataString = mapper.writeValueAsString(metadata);
+      metadataClone = mapper.readValue(metadataString, CommonMetadata.class);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    
+    ConversationHistoryEntry entry = new ConversationHistoryEntry();
+    entry.setMessageMap(new LinkedHashMap<String,String>(voiceInput.getMessageAsMap()));
```

Comment:
Does this matter if it's a LinkedHashMap vs regular HashMap? Are we guaranteed order on the incoming json that we parsed to get this? Does order even matter to us? I'm asking more for clarification rather than as a complaint.